src/compress/deflate/writer.cr

```diff
@@ -43,14 +43,17 @@ class Compress::Deflate::Writer < IO
   end
 
   # See `IO#write`.
-  def write(slice : Bytes) : Nil
+  def write(slice : Bytes) : UInt64
     check_open
 
-    return if slice.empty?
+    return 0u64 if slice.empty?
 
     @stream.avail_in = slice.size
     @stream.next_in = slice
     consume_output LibZ::Flush::NO_FLUSH
+
+    # CHECK: should we return the bytes written in the consume_output?
```

Comment:
I think `write` should always return the bytes sent to that `IO`, so the implementation is correct and this comment can be deleted.